File: ptv-mobile/plugins/pacc-native/src/main/java/com/potatotv/pacc/mobile/plugin/PaccPushPlugin.java
package com.potatotv.pacc.mobile.plugin;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

/**
 * 移动端推送插件（设计文档 §6.2.6，Android 走 FCM）。
 * 仅负责把 FCM token 上报给后端（走 PACC API），实际离线触达 / 红屏联动由服务端下发处理。
 */
@CapacitorPlugin(name = "PaccPush")
public class PaccPushPlugin extends Plugin {

    private String lastToken;

    @PluginMethod
    public void registerToken(PluginCall call) {
        String token = call.getString("token");
        if (token == null || token.isEmpty()) {
            call.reject("缺少 FCM token");
            return;
        }
        lastToken = token;
        // 上报到后端：此调用通过 PACC API 完成（HTTPS 会话），插件仅做本地暂存中转。
        JSObject ok = new JSObject();
        ok.put("registered", true);
        ok.put("token", token);
        call.resolve(ok);
    }

    @PluginMethod
    public void getToken(PluginCall call) {
        JSObject ok = new JSObject();
        ok.put("token", lastToken == null ? "" : lastToken);
        call.resolve(ok);
    }
}

File: ptv-mobile/ios/App/App/PaccPushPlugin.swift
import Capacitor
import Foundation

/// 推送插件（iOS APNs）。
/// 对应设计文档 §6.2.6：仅把 APNs deviceToken 中转给前端，由前端上报后端。
/// 方法签名与 Android PaccPushPlugin 对齐（registerToken/getToken）。
@objc(PaccPushPlugin)
public class PaccPushPlugin: CAPPlugin {

    private static var lastToken: String = ""

    /// 前端从原生 push 事件拿到 token 后，会回调到本方法做本地暂存。
    @objc func registerToken(_ call: CAPPluginCall) {
        guard let token = call.getString("token"), !token.isEmpty else {
            call.reject("缺少 APNs token")
            return
        }
        PaccPushPlugin.lastToken = token
        call.resolve(["registered": true, "token": token])
    }

    @objc func getToken(_ call: CAPPluginCall) {
        call.resolve(["token": PaccPushPlugin.lastToken])
    }

    /// 供 AppDelegate 注入 APNs token。
    static func capture(token: String) {
        lastToken = token
    }
}